### DISCUSSION TITLE
[RFC 1082 Discussion] - `uint` type abbreviation for FSharp.Core

### DISCUSSION THREAD

**Discussion by cartermp** [fsharp/fslang-design#423] [Language and core library RFC discussions]
This is the discussion issue for RFC 1082.

RFC: https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1082-uint-type-abbreviation.md
Suggestion: https://github.com/fsharp/fslang-suggestions/issues/818

**Comment by abelbraaksma** [comment]
The RFC doesn't mention the impact on the cast function. As mentioned in the discussion, I think the `uint()` function should be introduced as an alias for `uint32()`. Otherwise it's going to be rather odd that you must write this:

```f#
let f x: uint = uint32 x
```

Instead of this:

```f#
let f x: uint = uint x
```

It would also be nice if you could pair the other remaining primitives similarly, but I understand if that should go into a separate suggestion.

**Comment by cartermp** [comment]
Thanks @abelbraaksma - updating. I don't think we'd do that for the other primitives here, that would be another suggestion.

**Comment by varon** [comment]
PLEASE NO! This is going to be incredibly confusing to have `uint` vs `unit`.

**Comment by Happypig375** [comment]
PLEASE YES! This is going to be incredibly helpful to write `uint` vs `uint32`.

**Comment by davidglassborow** [comment]
Would a type alias not currently work ?

**Comment by cartermp** [comment]
@davidglassborow it is implemented as a type alias + casting functions in the current design. I wouldn't expect that to differ since that's how `int` is done.

**Comment by abelbraaksma** [comment]
> Would a type alias not currently work ?

That would alias the type, not the cast function.

**Comment by krauthaufen** [comment]
Hey, personally I'm not a huge fan of the many existing aliases (sbyte/int8/etc.) but when doing this how about short/ushort/long/ulong?

**Comment by Happypig375** [comment]
`uint` already exists as a concept today in F#: `uint8`, `uint16`, `uint32` and `uint64`.
In contrast, `short` and `long` do not exist in F# yet.

**Comment by abelbraaksma** [comment]
I personally find `short/int/long` confusing, since historically, and even presently in C/C++, `int` can mean 32 or 64 bit, `long` can mean 32 and 64 bit too. Sometimes it depends on word size, and a word can be 16, 32 or 64 bits.

Just to say that I find clarity in the way F# chose the names it did. The only one to remember is `(u)int`, the rest go by their bit-size-suffix. Adding more variants doesn't add coding clarity (and I'm aware we have both `sbyte` and `int8`, `float` and `double` and others, I wish there weren't, but that's a fait accompli).

**Comment by cartermp** [comment]
It's worth noting that there may be some more numeric primitives down the road in .NET. This would probably require corresponding type abbreviations in FSharp.Core.

**Comment by abelbraaksma** [comment]
I thought I saw an issue about a request for some new primitives like `unativeint` and the like.

**Comment by abelbraaksma** [comment]
I think this can be closed? The implementation is complete: https://github.com/dotnet/fsharp/pull/8185.

Note that a comment in the PR specifically says that this doesn't have y wait until F# 5.0, but that this thread is marked for the  F# 5.0 milestone.

**Comment by cartermp** [comment]
@abelbraaksma i'll likely do a round of closing later this year.
